igvm: Add SEV and SEV_ES platform types

roy-hopkins opened the pull request:
Currently, only SEV-SNP is supported as a platform type for SEV guests. However, there is no reason why IGVM cannot be used to configure both SEV and SEV-ES.

This commit adds two new platform types: `IgvmPlatformType::SEV` and `IgvmPlatformType::SEV_ES` which allow IGVM files to specifically target these isolation technologies.

Unlike SEV, SEV-ES allows the VP context to be specified. The structure for defining the context is binary compatible with SEV-SNP, therefore the SEV-ES platform type reuses IgvmDirectiveHeader::SnpVpContext to define the VMSA.

chris-oo commented on igvm_defs/src/lib.rs:
```diff
@@ -375,6 +375,10 @@ pub enum IgvmPlatformType {
     /// Intel TDX.
     TDX = 0x03,
+    /// AMD SEV.
+    SEV = 0x04,
```
hmm is it possible to gate these under the unstable feature since they're part of V2? Otherwise if we can't, it's probably okay.

chris-oo replied:
thanks!

chris-oo commented:
Is the initial CPU state for SEV settable by the host or is it a well-defined starting location?

roy-hopkins commented:
> Is the initial CPU state for SEV settable by the host or is it a well-defined starting location?

The CPU state is outside the protection model for SEV but it can be configured by the host, so I suppose we could define this using IGVM although it does not form part of any guest measurement. We could just allow the SEV-ES/SEV-SNP initial CPU state to also be used for SEV if you think it is appropriate.

chris-oo commented:
I wonder if it makes sense to use the NATIVE CPU state type then, since unlike TDX there's no requirement to be in a specific state? That at least makes more sense than the SNP VMSA page, since there's no measurement. What do you think?

roy-hopkins commented:
> I wonder if it makes sense to use the NATIVE CPU state type then, since unlike TDX there's no requirement to be in a specific state? That at least makes more sense than the SNP VMSA page, since there's no measurement. What do you think?

Yes, I think that makes sense. I've updated the description of `IgvmNativeVpContextX64` to include this as well as set the VP context type for SEV to `IgvmNativeVpContextX64` and updated the commit description.

chris-oo commented:
Changes LGTM.
